ProcessLib/ThermoMechanicalPhaseField/CreateThermoMechanicalPhaseFieldProcess.cpp

```diff
@@ -0,0 +1,277 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2018, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "CreateThermoMechanicalPhaseFieldProcess.h"
+
+#include "MaterialLib/SolidModels/CreateLinearElasticIsotropic.h"
+#include "MaterialLib/SolidModels/LinearElasticIsotropicPhaseField.h"
+#include "ProcessLib/Output/CreateSecondaryVariables.h"
+
+#include "ThermoMechanicalPhaseFieldProcess.h"
+#include "ThermoMechanicalPhaseFieldProcessData.h"
+
+namespace ProcessLib
+{
+namespace ThermoMechanicalPhaseField
+{
+template <int DisplacementDim>
+std::unique_ptr<Process> createThermoMechanicalPhaseFieldProcess(
+    MeshLib::Mesh& mesh,
+    std::unique_ptr<ProcessLib::AbstractJacobianAssembler>&& jacobian_assembler,
+    std::vector<ProcessVariable> const& variables,
+    std::vector<std::unique_ptr<ParameterBase>> const& parameters,
+    unsigned const integration_order,
+    BaseLib::ConfigTree const& config)
+{
+    //! \ogs_file_param{prj__processes__process__type}
+    config.checkConfigParameter("type", "THERMO_MECHANICAL_PHASE_FIELD");
+    DBUG("Create ThermoMechanicalPhaseFieldProcess.");
+
+    INFO("Solve the coupling with the staggered scheme,"
+         "which is the only option for TM-Phasefield in the current code");
+
+    // Process variable.
+
+    //! \ogs_file_param{prj__processes__process__THERMO_MECHANICAL_PHASE_FIELD__process_variables}
+    auto const pv_config = config.getConfigSubtree("process_variables");
+    ProcessVariable* variable_T;
+    ProcessVariable* variable_ph;
+    ProcessVariable* variable_u;
+    std::vector<std::vector<std::reference_wrapper<ProcessVariable>>>
+        process_variables;
+    int mechanics_related_process_id = 0;
+    int phase_field_process_id = 0;
+    int heat_conduction_process_id = 0;
+
+    using namespace std::string_literals;
+        for (auto const& variable_name :
```

Comment:
Need to use
```
        auto per_process_variables = findProcessVariables(
            variables, pv_config,
            {//! \ogs_file_param_special{prj__processes__process__THERMO_MECHANICAL_PHASE_FIELD__process_variables__temperature}
             "temperature",
             //! \ogs_file_param_special{prj__processes__process__THERMO_MECHANICAL_PHASE_FIELD__process_variables__phasefield}
             "phasefield",
             //! \ogs_file_param_special{prj__processes__process__THERMO_MECHANICAL_PHASE_FIELD__process_variables__displacement}
             "displacement"});
```
(as that already deleted for the monolithic scheme)
instead of this one. Otherwise, these parameters can not be documented.

Reply:
see my PR to the branch of this PR for changes.

Reply:
Need to apply clang-format.

Reply:
changes have been finished.